### ISSUE TITLE
Bug: Ionic serve showing old content

### ISSUE THREAD

**Issue by ihadeed** [driftyco/ionic-app-scripts#825]
#### Short description of the problem:
Ionic serve/livereload displays old content after refreshing. When I make changes to my app, the build completes and browser refreshes, then I see old content. Sometimes the changes do not apply, and sometimes I see an old page that I already changed multiple times. 

What I do to resolve this is restart `ionic serve` or run `npm run build` and refresh the page manually. Clearing browser cache and refreshing page solves this issue some times, but not all the time. So I don't think it's a browser cache issue.

#### What behavior are you expecting?
When the browser refreshes after a new build, it should display new updates.

**Steps to reproduce:**
This issue is really random.
1. Create new project
2. Run `ionic serve` or `ionic run <platform> -l`
3. Keep making changes to your app
4. Eventually the problem will occur

**Which @ionic/app-scripts version are you using?**
1.1.4

**Comment by duydao** [comment]
any chance you're using a custom webpack config file?

**Comment by norpoint** [comment]
We are seeing the exact same thing. Typically:
1) Ionic serve
2) Make some syntax error like <buitton> instead of <button>
3) See error response
4) Fix syntax error
5) Change something unrelated to previous error, eg. updating a .json file
6) Error response from step 3 repeats.

It does not seem to be browser caching, as cache reset / browser switch does not change this. The only way to workaround issue is to stop and rerun ionic serve.

We do not have local changes to webpack.

Ionic version 2.2.1. Package.json:

```
{
  "name": "ionic-hello-world",
  "author": "Ionic Framework",
  "homepage": "http://ionicframework.com/",
  "private": true,
  "scripts": {
    "clean": "ionic-app-scripts clean",
    "build": "ionic-app-scripts build",
    "ionic:build": "ionic-app-scripts build",
    "ionic:serve": "ionic-app-scripts serve"
  },
  "dependencies": {
    "@angular/common": "2.4.8",
    "@angular/compiler": "2.4.8",
    "@angular/compiler-cli": "2.4.8",
    "@angular/core": "2.4.8",
    "@angular/forms": "2.4.8",
    "@angular/http": "2.4.8",
    "@angular/platform-browser": "2.4.8",
    "@angular/platform-browser-dynamic": "2.4.8",
    "@angular/platform-server": "2.4.8",
    "@ionic/cloud-angular": "^0.11.0",
    "@ionic/storage": "2.0.0",
    "ionic-angular": "2.2.0",
    "ionic-native": "2.4.1",
    "ionicons": "3.0.0",
    "rxjs": "5.0.1",
    "sw-toolbox": "3.4.0",
    "zone.js": "0.7.2"
  },
  "devDependencies": {
    "@ionic/app-scripts": "1.1.4",
    "typescript": "2.0.9"
  },
  "cordovaPlugins": [
    "cordova-plugin-whitelist",
    "cordova-plugin-console",
    "cordova-plugin-statusbar",
    "cordova-plugin-device",
    "cordova-plugin-splashscreen",
    "ionic-plugin-keyboard"
  ],
  "cordovaPlatforms": [
    "ios",
    {
      "platform": "ios",
      "version": "",
      "locator": "ios"
    }
  ],
  "description": "lsb: An Ionic project"
}
```

**Comment by yuricamara** [comment]
I have exactly the same problem which @ihadeed described and I tried the same things

'Ionic serve' worked fine and suddenly this started to happen.

**Comment by Charlie-Hua** [comment]
Hi @duydao 
I am using an *extra* webpack config file to set environment configs.
Does that matter?
thanks.

**Comment by duydao** [comment]
@Charlie-Hua ionic uses filename from the webpack config to update the file on changes.

Ionic supports multiple entry points in our custom webpack.config.js:

```
  entry: {
    worker: workersDir + 'worker.ts',
    main: process.env.IONIC_APP_ENTRY_POINT
  },
```

but if we're trying to use a dynamic filename config, it won't update the bundle in ionic serve:
```
  output: {
    path: '{{BUILD}}',
    publicPath: 'build/',
    filename: '[name].js',
    devtoolModuleFilenameTemplate: ionicWebpackFactory.getSourceMapperFunction(),
  },
```

We can get around this fact by inlining the workers instead of generating separate files.

**Comment by artiomkichojal** [comment]
same issue

**Comment by danbucholtz** [comment]
This is fixed in `nightly`, will be available in `1.2.0` next week. `npm install @ionic/app-scripts@nightly`.

Thanks,
Dan

**Comment by ihadeed** [comment]
Thanks @danbucholtz !

**Comment by brettdorrans** [comment]
@danbucholtz 

This does not appear to be fixed after installing app-scripts@nightly for me. Same issue as before - old content is served, sometimes very old content (from several revisions ago).

**Comment by yuricamara** [comment]
Thanks! Should we update to @ionic/app-scripts@nightly or wait for next week release, @danbucholtz ?

**Comment by brettdorrans** [comment]
@danbucholtz 

So this bug is fixed in @nightly but it has introduced a new one, where sometimes SASS compilation fails completely and the app after rebuild displays without any style at all until a SASS file is saved again.

**Comment by ihadeed** [comment]
@brettdorrans you should open a new issue for that. I can confirm it's a bug.

**Comment by danbucholtz** [comment]
@brettdorrans,

Fixed in `1.2.0`. I was aware of the issue, it was on my todo list.

Thanks,
Dan

**Comment by brettdorrans** [comment]
@danbucholtz 

This issue persists in 1.2.2

**Comment by danbucholtz** [comment]
@brettdorrans,

How can I reproduce it?

Thanks,
Dan

**Comment by brettdorrans** [comment]
@danbucholtz 

Isn't easily reproducible. Seems to get worse with increasing app complexity (i.e. as you begin editing more pages/scss the effect is more noticeable). 

The steps to produce on my setup are:

1. Run ionic serve
2. Make change in Page1's markup and save (change is then reflected after reload)
3. Make change in Page1's scss and save (change is reflected after reload)
4. Make change in Page1's markup again, after reload displays markup as if change made in steps 2 and 3 did not occur.

**Comment by brettdorrans** [comment]
@danbucholtz 

Have narrowed it down to this:

I am using a custom ionic:serve in package.json so I can serve cordova in the browser.

"ionic:serve": "ionic-app-scripts serve --sourceMap source-map --iscordovaserve --wwwDir platforms/browser/www/ --buildDir platforms/browser/www/build",

This fails to update properly on save because ionic-app-scripts isn't running ionic build for the browser platform (perhaps it did this in an earlier version?)

I'm unable to use the standard ionic serve because it causes cordova related errors throughout the application.

**Comment by JSRenaissance** [comment]
Having same problem, I've noticed that I can save a blank space to the file that is cached to force it back into the build....

**Comment by reaVen-** [comment]
Same problem here. Even if i restart ionic serve an old version of the page shows up

**Comment by vgb1993** [comment]
Same problem here, suddently a bug appears because Ionic it's serving a old version of a file. I have to go to the file where de error is present, press save and sometimes it's gone. This has begin happening quite often since the last CLI update.  It's becoming **very annoying** to be honest. Some times I have to save around 4 different files to somehow get the latest version of the document. 

Ionic Framework: 3.7.1
Ionic App Scripts: 3.0.0
Angular Core: 4.4.3
Angular Compiler CLI: 4.4.3
Node: 6.11.1
OS Platform: macOS Sierra
Navigator Platform: MacIntel
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36

**Comment by tobbbe** [comment]
For me this only happens on Mac. Not windows.

**Comment by ihadeed** [comment]
I can confirm that this issue occurs on both Mac and Windows. I can tell it's happening when things stop making sense 😄 (I edit views/typescript and start seeing reverse progress).

The solution that works 100% of the time is to clear cache & restart `ionic serve`.

**Comment by alvarocasta** [comment]
same issue here. Exactly as described on main post.

@ionic/cli-utils  : 1.18.0
ionic (Ionic CLI) : 3.18.0
cordova (Cordova CLI) : 7.0.1
@ionic/app-scripts : 3.0.1
Cordova Platforms  : android 6.3.0 ios 4.5.2
Ionic Framework    : ionic-angular 3.8.0
ios-deploy : 1.9.1
ios-sim    : 5.0.6
Node       : v6.10.3
npm        : 5.5.1
OS         : macOS Sierra
Xcode      : Xcode 9.1 Build version 9B55

**Comment by brugh** [comment]
I have the same issue. running on Linux UB16.04LTS: 
ionic -v
3.18.0
cordova -v
7.1.0
npm -v
5.4.2
node -v
v6.10.2

**Comment by tobbbe** [comment]
Yep happens om both Windows and Mac for me now

**Comment by pdescham49** [comment]
WOW.. This is crazy. 

So I was making a provider "UserData" and decided to revert the file back to my old provider "AuthData"

Now on compile I get this: 

Runtime Error
Can't resolve all parameters for UserData: ([object Object], [object Object], ?).
Stack
```
Error: Can't resolve all parameters for UserData: ([object Object], [object Object], ?).
    at syntaxError (http://localhost:8100/build/main.js:80215:34)
    at CompileMetadataResolver._getDependenciesMetadata (http://localhost:8100/build/main.js:93552:35)
    at CompileMetadataResolver._getTypeMetadata (http://localhost:8100/build/main.js:93420:26)
    at CompileMetadataResolver._getInjectableMetadata (http://localhost:8100/build/main.js:93406:21)
    at CompileMetadataResolver.getProviderMetadata (http://localhost:8100/build/main.js:93696:40)
    at http://localhost:8100/build/main.js:93625:49
    at Array.forEach (native)
    at CompileMetadataResolver._getProvidersMetadata (http://localhost:8100/build/main.js:93586:19)
    at CompileMetadataResolver.getNgModuleMetadata (http://localhost:8100/build/main.js:93241:50)
    at JitCompiler._loadModules (http://localhost:8100/build/main.js:104305:66)
```

If I open up 'http://localhost:8100/build/main.js' I can see the reference. 

```var UserData = (function () {
    function UserData(events, storage, af) {
        this.events = events;
        this.storage = storage;
        this.af = af;
        this.HAS_LOGGED_IN = 'hasLoggedIn';
        this.fireAuth = __WEBPACK_IMPORTED_MODULE_4_firebase___default.a.auth();
    }
```

If I do a recursive search in my /src folder for "UserData" I get zero results. 

I've since deleted the 'www/build' folder and then started 'ionic server' again and this UserData Method still getting packed in webpack.. Even though it's nowhere to be found in /src/

Where is this coming from? I suspect the webpack has a cache or something? 

ionic -v
3.18.0

cordova -v
7.1.0

npm -v
3.10.10

node -v
v7.2.1

**Comment by pdescham49** [comment]
Soooooooo It appears that the source of the issue was in the Chrome APP CACHE. / Cache Storage. (under application) tab.

**Comment by tobbbe** [comment]
You've found a solution?

**Comment by alvarocasta** [comment]
Somehow the port 8100 seems corrupt on my case. I just run "ionic serve --port 8101" and this cache problem seems to be gone.
Not sure if this comes from Chrome or Webpack, so if someone finds out, please share.

**Comment by pdescham49** [comment]
@tobbbe I went into chrome debugger and clicked on the "application" tab and then click on the "clear storage" button. Might as well clear browser history as well. and restart the browser. 

Changing the port would also fix the issue since the cache is based on the url. ;) @alvarocasta

**Comment by NorthMcCormick** [comment]
So I don't know if this will help but I see this issue specifically on pages where I am importing this library like this: `import * as Plotly from 'plotly.js';` 

Ionic info:
```cli packages: (/usr/local/Cellar/node@6/6.11.4_1/lib/node_modules)

    @ionic/cli-utils  : 1.19.0
    ionic (Ionic CLI) : 3.19.0

global packages:

    cordova (Cordova CLI) : 7.0.1 

local packages:

    @ionic/app-scripts : 3.0.0
    Cordova Platforms  : ios 4.4.0
    Ionic Framework    : ionic-angular 3.7.1

System:

    ios-deploy : 1.9.0 
    ios-sim    : 5.0.11 
    Node       : v6.11.4
    npm        : 3.10.8 
    OS         : macOS High Sierra
    Xcode      : Xcode 9.1 Build version 9B55 

Environment Variables:

    ANDROID_HOME : not set

Misc:

    backend : pro```

**Comment by chris08002** [comment]
Are you using service workers? Just faces the issue that ionic serve does rebuild on every code change but the app doesn't change. After deleting all application cache and it worked.

This gave me the right hint: https://stackoverflow.com/questions/42372872/ionic-serve-loading-wrong-app

**Comment by alexboisselle-rd** [comment]
This is absolutely not fixed. Using latest and still getting this issue.

**Comment by tekkenking** [comment]
Still having the same issue here..

**Comment by matheusdavidson** [comment]
This still happens with latest version. This is an old issue, i remember to have this since the first time i used ionic 2.

**Comment by ZacharyCody** [comment]
Bug is still apparent

**Comment by tobinbc** [comment]
Definitely still a problem, but as mentioned, often adding a space to one of the 'wrong' files and saving will bring it back to life.

**Comment by JRobber** [comment]
This issue is still present as is affecting deployment.  

Build an APK following the steps here : https://ionicframework.com/docs/intro/deploying/.
Push it to the google play console and start an alpha build.  
Load the app through the alpha build on a fresh device.
Force kill and reload the app a few times.
Within 10 tries we're seeing an old menu path that was removed 8 builds and 2 months ago.
It's completely mind-boggling how code that has been deleted from git 2 months ago shows up in the latest build, sometimes, after killing the app a few times.

**Comment by gual** [comment]
Experiencing what @JRobber states.

**Comment by ndcunningham** [comment]
Having the same problem as OP

Can this be reopened i don't think the issue is closed.

**Comment by tunde-dovenet** [comment]
Experiencing same issue as @JRobber. There was no issue when working with ionic serve. The issue started after deploying to Google play store

**Comment by gydammin** [comment]
Same problem happing, ionic serve do the job but content doesn't get refreshed. Closing and opening chrome, even in in incognito mode, does not change anything. Need to rerun ionic serve then re-open browser in incognito mode then I get content updated.
Previously this refresh problem was solved by working in incognito mode, but know it does not do the trick anymore :/

**Comment by ValdoGhafoor** [comment]
Developping with Ionic is a pain with that bug. Just imagine that we have to refresh around 200 times an hour ... Please fix this

**Comment by tunde-dovenet** [comment]
I found out that while developing the app, I uploaded my app to Ionic Pro
about 2 months ago. I did not update it

When I published the app to app store, the old copy in ionic Pro update the
newer version.

What I did to resolve this issue is to delete everything in my Ionic Pro
and republish the app.

**Comment by matheusdavidson** [comment]
the problem starts to get worse when you separate everything in components.

**Comment by matheusdavidson** [comment]
The problem is worse when you use multiple components in one page, in my case everything is lazy loaded, even components.

**Comment by edisonmatoso** [comment]
Same issue here, and my problem is similar to @matheusdavidson.

Ionic v3.9.2
Ionic Scripts v3.1.8
Angular v5.0.3
Node v8.9.3
win10

**Comment by andy008** [comment]
Same issue here. A build will fail with an old error in Chrome. Hit Localhost in Firefox and it works. Constantly resolving with a mix of cache clearing and occasional restart of ionic serve. Very frustrating through as I find myself often chasing non-existent bugs.

**Comment by jcarloTA** [comment]
Same issue here. I have to return to my files and save them again to see the changes.

**Comment by toffebjorkskog** [comment]
Same here. If i for instance edit a components template file, i first need to save it, then it's .ts file, then  components.module.ts and lastly app.components.ts in order for the change to be visible in the app. Can it be related to lazy loading of pages.

**Comment by russgalloway** [comment]
ionic: 3.20.0
cordova: 8.0.0
npm: 5.5.1
node: 8.9.0
ios-sim: 5.0.6
ios platform: 4.5.4
os: macOS Sierra 10.12.6

I've been experiencing the same ever since I started working with Ionic heavily a month or so ago.

For instance, I'll create a function in a component and reference it in the template as a button click handler.  Later I might rename the function in the component and update the template reference.  

After several template/component changes I'll make an innocuous change to the template (i.e. fix a spelling mistake in a span) and it'll throw an error stating that the old click handler function name doesn't exist in the template's component source.

Most of the time I can resolve it by adding/removing a space in the template's component source and save the file forcing another build cycle.  When that stops working I have to stop and re-run `ionic cordova run ios --livereload` -- which happens several times a day.

Since I've seen it mentioned above -- I do have a 'custom' webpack config in order to work  around a nodejs vs browser issue with TypeORM (https://github.com/typeorm/ionic-example).  I don't think this is relevant to this (build cache?) issue because I experienced this issue when I was developing the UI in Chrome -- before including TypeORM and switching to the iOS simulator.

Hope this helps.

**Comment by EugenRakhimov** [comment]
Same issue here. Updated an application, builded a version with ionicPro, and still old content is showing on the phone, even after I deleted version and installed from google again.

**Comment by stellah57** [comment]
Me too experiencing the same issue. It's so frustrating that I have to go through all previously edited files and press Cmd+s in order to bring the latest version of the files. Sometimes even error on non-existing method shows up. Been diving into ionic for about 2 months now, it's toxic to the bone due to this bug.

**Comment by tejon-melero** [comment]
I am experiencing the same issue. Is this already accepted as an official bug/issue? It really is a huge issue when developing.
`cli packages: (/home/matyas/.nvm/versions/node/v9.3.0/lib/node_modules)
    @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 3.20.0
global packages:
    cordova (Cordova CLI) : 8.0.0 
local packages:
    @ionic/app-scripts : 3.1.5
    Cordova Platforms  : none
    Ionic Framework    : ionic-angular 3.9.2
System:
    Node : v9.3.0
    npm  : 5.5.1 
    OS   : Linux 4.13
Environment Variables:
    ANDROID_HOME : not set
Misc:
    backend : pro
`

**Comment by blaforet** [comment]
Hi 
I have the same issue and it's been frustrating.   
## **The only solutions that i found so far is to use FIREFOX**
so far t's been stable and does reload correctly after a save.

with Chrome ionic simply does not reload correctly and even load a VERY OLD Version.  Clearing the cache every time is a solution but time consuming.
I think i have a recent version of almost everything, do not have this issue with Chrome and Angular project
cli packages: (C:\Angular\Ionic\MT-Admin\node_modules)
    @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 3.20.0

global packages:
    cordova (Cordova CLI) : 8.0.0

local packages:
    @ionic/app-scripts : 3.1.7-201801172029
    Cordova Platforms  : android 6.3.0
    Ionic Framework    : ionic-angular 3.9.2

System:
    Node : v9.5.0
    npm  : 5.6.0
    OS   : Windows 10
Environment Variables:
    ANDROID_HOME : not set
Misc:
    backend : pro

I Hope it will help some users

**Comment by stansongman** [comment]
This just started happening to me in Chrome.  I have to bring up the developer tools, go to Application, right-click on ionic-cache and select Delete.  Then I refresh the window.

**Comment by andy008** [comment]
Stop / start ionic serve fixes the problem.

Often, just re-saving app.module.ts can trigger a build that fixes the problem - but not *always*.

This is extremely frustrating and wastes a lot of time! Is there really no fix coming on this guys?

**Comment by TheAlexPorter** [comment]
I found a solution that worked, maybe this will help some of you here.

Roll back your app scripts to version 1.3.1. Even though it isn't the latest version, everything will still work.

`npm uninstall @ionic/app-scripts`

`npm install --save-dev @ionic/app-scripts@1.3.1`

This cleared the issue right away for me. I hope this helps while we wait for the current scripts to be sorted!

**Comment by andy008** [comment]
@TheAlexPorter Tried your solution, but got an ERR! code ELIFECYCLE, errno 1 installing 1.3.1

**Comment by TheAlexPorter** [comment]
@andy008 Shoot! I'd hoped it would help. Here are my other dependencies, perhaps that will help? If not, I'd try changing your app scripts version around and see if you find a fix your dependencies version requirements.

"dependencies": {
    "@angular/animations": "5.2.9",
    "@angular/common": "5.2.9",
    "@angular/compiler": "5.2.9",
    "@angular/compiler-cli": "5.2.9",
    "@angular/core": "5.2.9",
    "@angular/forms": "5.2.9",
    "@angular/http": "5.2.9",
    "@angular/platform-browser": "5.2.9",
    "@angular/platform-browser-dynamic": "5.2.9",
    "@angular/tsc-wrapped": "^4.4.6",
    "@ionic-native/core": "4.6.0",
    "@ionic-native/google-analytics": "^4.6.0",
    "@ionic-native/in-app-browser": "^4.6.0",
    "@ionic-native/screen-orientation": "^4.6.0",
    "@ionic-native/splash-screen": "4.6.0",
    "@ionic-native/status-bar": "4.6.0",
    "@ionic/pro": "1.0.20",
    "@ionic/storage": "^2.1.3",
    "cordova-android": "6.3.0",
    "cordova-ios": "4.5.4",
    "cordova-plugin-device": "^2.0.1",
    "cordova-plugin-google-analytics": "^1.8.3",
    "cordova-plugin-inappbrowser": "^2.0.2",
    "cordova-plugin-ionic-keyboard": "^2.0.5",
    "cordova-plugin-ionic-webview": "^1.1.16",
    "cordova-plugin-screen-orientation": "^3.0.1",
    "cordova-plugin-splashscreen": "^5.0.2",
    "cordova-plugin-statusbar": "git+https://github.com/apache/cordova-plugin-statusbar.git",
    "cordova-plugin-whitelist": "^1.3.3",
    "cordova-sqlite-storage": "^2.3.0",
    "es6-promise-plugin": "^4.2.2",
    "ionic-angular": "3.9.2",
    "ionicons": "3.0.0",
    "rxjs": "5.5.8",
    "sw-toolbox": "3.6.0",
    "zone.js": "0.8.20"
  },
  "devDependencies": {
    "@ionic/app-scripts": "^1.3.1",
    "typescript": "~2.6.2"
  },

**Comment by stellah57** [comment]
Weeks ago I've decided that I had enough on this frustrating issue. I moved
away to react and xamarin.. It's much better time investment to learn new
stuff rather than chasing  around ghostly bugs  in my own codes.

**Comment by zen4raul** [comment]
I have faced similar issue - is there a config set option to disable cache for all views for development run debugging ? Also i faced caching of httpclient get results - does anybody know if there any option to also disable caching of http get results ?

**Comment by AndoniXV** [comment]
This is a very frustrating problem for me. Sometimes I can solve it by stopping/starting ionic serve. But other times I can spend like 20-30min deleting cache, stopping ionic serve, deleting the www folder, opening it on firefox, saving all the files 10 times and nothing works... And it can even happen with production builds too.

For me this only happens on Windows. On Mac it works perfectly, I don't have any problem there. I don't really know if I'm missing something or what, but sometimes it can get really frustrating when I'm working on my PC at home.

**Comment by zen4raul** [comment]
also has anybody experienced httpclient (angular) throwing stale/cached results when using ionic in debug -lcd mode ? any workaround to always get fresh http get results ?

**Comment by Ross-Rawlins** [comment]
Any new news on this?

**Comment by jcyh0120** [comment]
I am facing this problem too.

**Comment by ludobit** [comment]
Same bug! Every time i refresh my app on the android device I lose some new code wich is reverted to the old one. 

`
 "dependencies": {
    "@angular/animations": "5.2.10",
    "@angular/common": "5.2.10",
    "@angular/compiler": "5.2.10",
    "@angular/compiler-cli": "5.2.10",
    "@angular/core": "5.2.10",
    "@angular/forms": "5.2.10",
    "@angular/http": "5.2.10",
    "@angular/platform-browser": "5.2.10",
    "@angular/platform-browser-dynamic": "5.2.10",
    "@ionic-native/camera": "4.3.3",
    "@ionic-native/core": "4.7.0",
    "@ionic-native/splash-screen": "4.7.0",
    "@ionic-native/status-bar": "4.7.0",
    "@ionic/storage": "2.1.3",
    "@ngx-translate/core": "8.0.0",
    "@ngx-translate/http-loader": "^2.0.0",
    "cordova-android": "7.0.0",
    "cordova-plugin-device": "^2.0.2",
    "cordova-plugin-ionic-keyboard": "^2.0.5",
    "cordova-plugin-ionic-webview": "^1.1.19",
    "cordova-plugin-splashscreen": "^5.0.2",
    "cordova-plugin-whitelist": "^1.3.3",
    "ionic-angular": "3.9.2",
    "ionicons": "3.0.0",
    "lodash": "^4.17.10",
    "rxjs": "5.5.10",
    "sw-toolbox": "3.6.0",
    "zone.js": "0.8.26"
  },
  "devDependencies": {
    "@ionic/app-scripts": "3.1.9",
    "typescript": "~2.6.2"
  }
`

**Comment by pdanysz** [comment]
I have the similar problem, but on Chrome for me is enough to click Shift-Ctrl-R to reload and clear Chrome cache. BUT :)

When I deploy app to mobile (device) I have the same issue (as example):
- l'm login in to app and select one Event Item from the list
- After selecting, it download posts for selected event 
- logout from app
- I do not close the app, just going back to login page
- I login again, select different Event (different ID in /api path) and it shows me the same posts as for previous  WFT?!?!? 

And yes, I checked it is not a problem with my app , api, and server side cache :), because it uses differend api path:
```
/api/events/<id....>/posts
/api/events/<id22222>/posts
```

Exactly the same I have when "ionic serve" I do for web, and I'm sure that is caused by ionic-cache , but how to clear it from app side ?

```
ionic: 3.20.0
cordova: 8.0.0
npm 5.6.0
node: 8.9.1
```

**Comment by geass619** [comment]
The issue still exists, I am using a windows machine.
Will the problem be solved if used in Firefox !?
The ionic team should work on fixing it ASAP or provide a solution, we cannot go on doubting whether the code is working or not on every new line.
With no solution, you guys are making us move to another option like react...

**Comment by santhoshopus** [comment]
Yes, the same problem exists & its very frustrating. 
It usually gets solved by completely stopping ionic serve and running ionic serve again sometimes its takes several attempts to resolve it. But it again it re-appears after making changes to the files. Wasting lots of time. I was doing my first project on ionic, and still regretting my decision to move to ionic due to various reasons and this is among the primary reason for it.

```
cli packages: (/usr/local/lib/node_modules)
    @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 3.20.0

global packages:
    cordova (Cordova CLI) : 8.0.0 

local packages:
    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : android 7.0.0 browser 5.0.3 ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2
```

**Comment by tejon-melero** [comment]
@danbucholtz could you please reopen this bug since a lot of people are proving here that this issue still exists? Is there any way how we could help you to isolate this specific behaviour? Or is there any way to disable the ionic cache in general?

**Comment by santhoshopus** [comment]
This problem is driving me nuts. It throws random errors for no reason, that gets fixed upon stopping and serving again. 

For example I get the the following error, randomly. Sometimes some other errors.
`TypeError: __WEBPACK_IMPORTED_MODULE_6__ionic_storage__.a.forRoot is not a function
    at http://localhost:8100/build/main.js:1475:92
    at Object.498 (http://localhost:8100/build/main.js:1511:2)
    at __webpack_require__ (http://localhost:8100/build/vendor.js:55:30)
    at Object.493 (http://localhost:8100/build/main.js:1370:70)
    at __webpack_require__ (http://localhost:8100/build/vendor.js:55:30)
    at webpackJsonpCallback (http://localhost:8100/build/vendor.js:26:23)
    at http://localhost:8100/build/main.js:1:1`

**Comment by vjoao** [comment]
Same problem. This has a huge impact on productivity.

**Comment by pdescham49** [comment]
If this is the same issue I think it is..

The work around is to specify a new port and restart. --port=8888

Paul.

On Wed, Jun 6, 2018 at 10:41 AM, Victor Nascimento <notifications@github.com

**Comment by RaschidGithub** [comment]
> The work around is to specify a new port and restart. --port=8888

@pdescham49 So you do this each time or just doing it once fixes the problem?

**Comment by pdescham49** [comment]
When I notice the content is cached.. I'll stop it restart it on a new port
and it will work again.. sometimes for days, sometimes for hours.

**Comment by pdescham49** [comment]
In my experience you don't even need to delete the www folder. But yes that
will do as well.

**Comment by eduardo-mattos** [comment]
Same problem here.

1-Ionic serve
2-Make some syntax error like <buitton> instead of <button>
3-See error response
4-Fix syntax error
5-Change something unrelated to previous error, eg. updating a .json file
6-Error response from step 3 repeats.

**Comment by vjoao** [comment]
Solved it: I don't use Ionic anymore.

**Comment by walid-deutscher** [comment]
a big problem
really it is a big problem
we need evrey time to clear cache in divice

**Comment by walid-deutscher** [comment]
the probleme ist 100% solved by me
when i use http  Plugin  seem this problem


i see old  data http  not updated  
............................
first step yu need to  host  json.php  in domain free or paid like do yu want
www.yourdomain.com/json.php
 your code
<?php
header("Access-Control-Allow-Origin: *");
?>
<meta http-equiv="refresh" content="4" />
<?php

	
	$url = ' your json lik  the sourcee'; 
$data = file_get_contents($url);  
$characters = json_decode($data); 

$myJSON = json_encode($characters);

echo $myJSON;

?>

 in ionic in http plugin you need to use this


dz=new Date();  // very importante

	  this.http.get('www.yourdomain.com/json.php.php?dz='+dz).map(res => res.json()).subscribe(data => {

consol.log(data); 
	 );

**Comment by tejon-melero** [comment]
It might be helpful to post to this issue thread as well (since this bug is officially closed). 
It is the same issue that was reopened again
https://github.com/ionic-team/ionic-app-scripts/issues/1432
one possible workaround seems to be changing the service worker settings in the chrome developer settings:
https://grzegorowski.com/ionic-framework-live-reloading/

**Comment by Vaibhavi24** [comment]
I experienced the same issue, it keeps building the last 3rd iteration.

Is running ionic-serve is the only work around for this?

**Comment by clemishow** [comment]
I need to reload the server every time...

```
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 3.20.0

global packages:

    cordova (Cordova CLI) : 8.0.0

local packages:

    @ionic/app-scripts : 3.1.9
    Cordova Platforms  : ios 4.5.4
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 26.1.1
    ios-deploy        : 1.9.2
    ios-sim           : 6.1.2
    Node              : v8.10.0
    npm               : 5.6.0
    OS                : macOS High Sierra
    Xcode             : Xcode 9.4 Build version 9F1027a

Environment Variables:

    ANDROID_HOME : /Users/dev/Library/Android/sdk

Misc:

    backend : pro
```

**Comment by matheusdavidson** [comment]
upgrade to ionic 4 if you can, the problem will go away as you won't be using ionic-app-scripts =]

**Comment by airstep** [comment]
So, anyone can handle browser to clear cache. It is usefull if there is new version of the app bumped but older content still present...

**Comment by m--c** [comment]
I have found that in my case, this happens when I use a templateUrl to an html which is in another folder.
for instance, in` welcome.ts`, I used 
`templateUrl: '../pages/menu/menu.html'`
If i refer to the real welcome.ts, it does incliiude the changes. It seems to have something to do with not realizing that other file is in the project structure

**Comment by psimondk** [comment]
Just upgraded to ionic4 and created a sidemenu app - same problem. Only solution is to manually clear the cache in chrome and reload ... arghhh 👎 👎 👎

**Comment by finzaiko** [comment]
Its Ok I just re-run `ionic serve --no-open` resolved me for while

**Comment by EdgarSalazar** [comment]
If you're using custom webpack.config (as i were), following these steps worked for me:

https://github.com/gshigeto/ionic-environment-variables

I just recreated my whole webpack.config based on this tutorial, and it worked fine and sweet.

**Comment by EdgarSalazar** [comment]
Actually, i started having it again. Then i noticed livereload suddenly started to work when i update my custom webpack file, and served the app again. Somehow, this are being cached. Updating webpack file (like putting a empty line behind everything) worked for me.

**Comment by EdgarSalazar** [comment]
UPDATE

Using anonymous window on Chrome seems to be the easiest way to work. Problem still exists on Ionic 3.9.2

**Comment by ptrahan** [comment]
@EdgarSalazar 
Thanks! Been pulling my hair. To me the best compromise for this bug is:
> ionic serve --no-open
And in chrome use an incognito window and open the url to your app. Live changes you make are seen as you would expect, did not need to clear cache or restart ionic serve.

I had abandoned Ionic last year after 6 months of time wasting on numerous issues. I come back a year later, giving another chance to this framework, but I'm already starting to regret...

**Comment by sfearl1** [comment]
For those coming here having issues w/ Ionic 3+ and iOS/Android build showing old views...

This was happening when I would build/run for iOS and Android.  My latest code would show up on first load and each subsequent load (after killing the app) on both platforms would show old component views.  I updated all npm/ionic/cordova packages, plugins, cli + confirmed all updates, cleared all cache that I could find, removed all build and plugin folders, rebuilt everything and still had no luck.  

I finally had to go in to Ionic Dashboard > Personal Apps and delete my Ionic Pro link/app.  I'm not sure how my builds were accessing my old code via Ionic Pro during the build phase, but deleting the link/app appears to have fixed the problem for me.

FWIW, I think I could've fixed the problem without deleting the Pro link by pushing each build to the Channel/branch that was connected to my Ionic Pro repo.  Maybe one of the Ionic guys can confirm this.
